client/src/components/DetailsCard/index.js

```diff
@@ -1,34 +1,91 @@
-import React from "react";
+import React, { Component } from "react";
 import PropTypes from "prop-types";
 
 import "./detailscard.css";
 
-const DetailsCard = ({ boardingTime, destination, origin, flightNo }) => (
-  <div className="center-div">
-    <div className="container-card-details">
-      <div className="labels-container">
-        <label className="details-label">Flight number</label>
-        <label className="float-style">{flightNo}</label>
-      </div>
-      <div className="labels-container">
-        <label className="details-label">From</label>
-        <label className="float-style">{origin}</label>
-      </div>
-      <div className="labels-container">
-        <label className="details-label">To</label>
-        <label className="float-style">{destination}</label>
-      </div>
-      <div className="labels-container">
-        <label className="details-label">Arrival Time</label>
-        <label className="float-style">{boardingTime}</label>
+class DetailsCard extends Component {
+  getHowManyhourToDepart(mytime) {
+    const dt1 = new Date();
+    const dt2 = new Date(mytime);
+    var diff = (dt2.getTime() - dt1.getTime()) / 1000;
+    diff /= 60 * 60;
+    return Math.abs(Math.round(diff));
+  }
+
+  render() {
+    const {
+      adminPanel,
+      boarding_time,
+      destination,
+      origin,
+      flight_no,
+      detailsTitle,
+      detailsInfoArray
+    } = this.props;
+
+    return (
+      <div>
+        {adminPanel ? (
```

Comment:
This switching between a large block of admin-related content and a large block of customer-related content inside render() is indicating to me that this might be better off as two different components instead of one component that can do both jobs.
I think the parts that they share is then mostly styling, which they could still share by both using the same css file and using the same classNames (like they already are, for example `center-div`).